next.clayui.com/content/docs/components/modal.mdx

```diff
@@ -29,6 +32,147 @@ You may want to compose your content and customize with your use cases or add yo
 
 <Modal />
 
+### useModal hook
+
+The `useModal` [hook](https://reactjs.org/docs/hooks-intro.html) is required to create a Modal component in Clay, it is responsible for communicating with ClayModal and controlling states to make animations, this is necessary because Modal needs animation when opening and closing so we need to delay calls of the `onClose` so you can safely control Modal's visibility or do other things.
+
+`useModal` returns a signature with `observer` and `onClose`:
+
+- **observer** - Observer is the communication channel that connects ClayModal with `useModal`, you can call it a tunnel.
+- **onClose** - Callback is called after Modal's close animation.
+
+```jsx
+const { observer, onClose } = useModal({
+    onClose: () => setVisible(false)
+});
+```
+
+<div class="clay-site-alert alert alert-warning">
+    To avoid future problems, do not use the information or change the values ​​of `observer`, it is considered a bad practice.
+</div>
+
+To prevent Modal from closing before animation never set its state directly in button events to close the modal instead call the `onClose` method returned by `useModal`.
```

Comment:
```suggestion
To prevent Modal from closing before animation always use the `onClose` method returned by `useModal`.
```

I think this should be clear and info info to convey that we want them to use the hook.